cmd/web-handlers.go

```diff
@@ -737,14 +738,24 @@ next:
 					return toJSONError(ctx, errAccessDenied)
 				}
 			}
-			_, replicateDel := checkReplicateDelete(ctx, getObjectInfoFn, args.BucketName, ObjectToDelete{ObjectName: objectName})
-			if replicateDel {
-				opts.DeleteMarkerReplicationStatus = string(replication.Pending)
-				opts.DeleteMarker = true
+			var (
+				replicateDel, hasLifecycleConfig bool
+				goi                              ObjectInfo
+				gerr                             error
+			)
+			if _, err := globalBucketMetadataSys.GetLifecycleConfig(args.BucketName); err == nil {
```

Comment:
Could you explain why we need lifecycle in web-handlers.RemoveObject() ?

Reply:
an object that has transitioned can still be deleted via regular `mc rm` instead of waiting for the lifecycle to kick in.To know if it is transitioned - GetObjectInfo is needed. Checking if lifecycle exists is only to see if we can avoid GetObjectInfo call